go/engine/deprovision.go

```
@@ -44,23 +44,41 @@ func (e *DeprovisionEngine) SubConsumers() []libkb.UIConsumer {
 	}
 }
 
-func (e *DeprovisionEngine) Run(ctx *Context) (err error) {
-	// Deprovision steps
-	// =================
-	// 1. If the user is logged in:
-	//   a) Revoke all the current device's keys.
-	//   b) Log out.
-	// 2. Delete all the user's secret keys!!!
-	// 3. Delete the user from the config file.
-	// 4. Db nuke.
-
-	// If the user to deprovision is currently logged in, we need to revoke
-	// their keys and then log out.
+// This function anticipates some error cases, but it will still return an
+// error if something unexpected goes wrong.
+func (e *DeprovisionEngine) attemptRevoke(ctx *Context) error {
 	isLoggedIn, _, err := IsLoggedIn(e, ctx)
 	if err != nil {
+		e.G().Log.Debug("DeprovisionEngine error checking login status: %s", err)
 		return err
 	}
-	if e.G().Env.GetUsername().Eq(e.username) && isLoggedIn {
+
+	// If the user we're deprovisioning isn't logged in, short-circuit the
+	// revoke and continue deprovision.
+	if !isLoggedIn || !e.G().Env.GetUsername().Eq(e.username) {
+		ctx.LogUI.Warning("User %s is not logged in, so we aren't revoking their keys on the server.", e.username)
+		ctx.LogUI.Warning("To do that yourself, run `keybase device remove %s`", e.G().Env.GetDeviceID())
+		ctx.LogUI.Warning("from another device.")
+		return nil
```

Maybe return an error and require a flag here.